forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-gateway/forgerock-openbanking-uk-aspsp-rs-gateway-server/src/test/java/com/forgerock/openbanking/aspsp/rs/wrappper/endpoints/PaymentsRequestPaymentIdEndpointWrapperTest.java

```diff
@@ -0,0 +1,135 @@
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.forgerock.openbanking.aspsp.rs.wrappper.endpoints;
+
+import com.forgerock.openbanking.am.config.AMOpenBankingConfiguration;
+import com.forgerock.openbanking.am.services.AMResourceServerService;
+import com.forgerock.openbanking.aspsp.rs.wrappper.RSEndpointWrapperService;
+import com.forgerock.openbanking.common.model.openbanking.forgerock.ConsentStatusCode;
+import com.forgerock.openbanking.common.model.openbanking.forgerock.FRPaymentConsent;
+import com.forgerock.openbanking.common.model.openbanking.v3_0.payment.FRDomesticConsent1;
+import com.forgerock.openbanking.constants.OIDCConstants;
+import com.forgerock.openbanking.exceptions.OBErrorException;
+import com.forgerock.openbanking.jwt.services.CryptoApiClient;
+import com.forgerock.openbanking.model.error.ErrorCode;
+import dev.openbanking4.spring.security.multiauth.model.authentication.PasswordLessUserNameAuthentication;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import static com.forgerock.openbanking.integration.test.support.JWT.jws;
+import static org.assertj.core.api.Assertions.*;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+public class PaymentsRequestPaymentIdEndpointWrapperTest {
+    private static PaymentsRequestPaymentIdEndpointWrapper wrapper;
+
+    @MockBean(name = "cryptoApiClient")
+    private CryptoApiClient cryptoApiClient;
+
+    @MockBean(name = "amOpenBankingConfiguration")
+    private AMOpenBankingConfiguration amOpenBankingConfiguration;
+
+    @MockBean(name = "amResourceServerService")
+    private RSEndpointWrapperService rsEndpointWrapperService;
+
+    @Before
+    public void setup() {
+        // setting required objects to the perform test
+        ReflectionTestUtils.setField(amOpenBankingConfiguration, "audiences", Arrays.asList("https://am.dev-ob.forgerock.financial:443/oauth2/auth"));
```

Comment:
If the object is mocked you shouldn't need to set the field

Reply:
I have found another way using mockito to avoid set the fields.